evap/evaluation/templates/base.html

```diff
@@ -85,41 +83,70 @@
+            var templateForm = $(".tomselecttr").last().clone();
 
             // later we use this class to give this to django-dynamic-formsets
-            // as template and to make sure it does not get select2'd
+            // as template and to make sure it does not get tomselect'd
             templateForm.addClass("form-template").css("display", "none");
             // for some reason django-dynamic-formsets does not hide the checkbox like it should
             templateForm.find(':checkbox').last().attr("type", "hidden");
             // For some reason, django validates this template if it's part of the form, so we insert the copy outside of the form.
-            templateForm.insertAfter($('.select2form'));
+            templateForm.insertAfter($('.tomselectform'));
 
-            // put this into a function for later re-use
-            applySelect2 = function(elem, parent = $(document.body)) {
-                elem.each(function() {
-                    $(this).select2({
-                        language: "{{ LANGUAGE_CODE }}",
+            // used in derived templates
+            const tagTomSelectOptions = {
+                create: true,
+                dropdownParent: document.createElement("div"),
+            };
+
+            applyTomSelect = function(elements, additionalOptions = {}) {
+                elements.forEach(function(element) {
+                    if(element.tomselect) {
+                        element.tomselect.destroy();
+                    }
+                    element.classList.remove("form-select");  // TomSelect applies their own matching classes / styles
+                    if(element.parentElement.classList.contains("is-invalid")) {  // hotfix for our wrongly placed is-invalid
+                        element.parentElement.classList.remove("is-invalid");
+                        element.classList.add("is-invalid");
+                    }
+                    const baseOptions = {
+                        createOnBlur: true,
                         placeholder: "{% trans 'Please select...' %}",
-                        minimumResultsForSearch: 15,
-                        minimumInputLength: $(this).children('option').length > 50 ? 3 : 0,
-                        allowClear: this.multiple,
-                        dropdownParent: parent
-                    });
+                        hidePlaceholder: true,
+                        render: {
+                            option_create: function( data, escape ){return '<div class="create">' + escape(data.input) + '</div>';},
+                            no_results: function( data, escape ){return '<div class="no-results">{% trans "No results found" %}</div>';},
+                        },
+                        plugins: {
+                            // "dropdown_input": {}, // # TODO: Do we want this? I think both are a bit weird, with and without
```

Comment:
@janno42 What's your opinion here?
* Without this plugin, the typing to search an entry, the typed text will be visible directly next to the current selected item, even on single-select widgets.
* With this plugin, the typed text is shown in the dropdown (which I think we want), but if no item was selected, the height of the field will break:
<img height=200 src="https://user-images.githubusercontent.com/13838962/168425461-afa93ffb-25c1-4830-8ae2-696e441da626.png"></img>

Maybe worth a bugreport with tomselect?

Reply:
The resized input field happens because they set the input field with the placeholder inside it to `display: None !important`. Setting it to `visibility: hidden` instead fixes it. We may be able to fix this in our CSS, and Tom Select would probably update their plugin if we issue a bug report.

Reply:
Opened issue 383 on tomselect, they committed a fix, so this should work in the next tomselect release. Then, we probably want to use the "dropdown_input" plugin?

Reply:
Their issue was fixed and they released 2.0.3. Updated the version in this PR.